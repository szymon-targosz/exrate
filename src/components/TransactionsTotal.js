import React from 'react';
import InfoBox from './UI/InfoBox';
import { generateAmountMsg } from '../helpers/transactions';

const TransactionsTotal = ({ total, rate, quantity }) => (
   <section className="total">
      <InfoBox
         description={`Total (${quantity} transactions)`}
         data={`${generateAmountMsg(total)} / ${generateAmountMsg(total * rate, 'pl')}`}
      />
   </section>
);

export default TransactionsTotal;
