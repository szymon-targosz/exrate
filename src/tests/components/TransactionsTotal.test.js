import React from 'react';
import TransactionsTotal from '../../components/TransactionsTotal';
import { shallow } from 'enzyme';

describe('COMPONENT - TransactionsTotal', () => {
   test('Should render TransactionsTotal correctly', () => {
      const wrapper = shallow(<TransactionsTotal total={583.07} rate={3.132321} quantity={7} />);
      expect(wrapper).toMatchSnapshot();
   });
});
