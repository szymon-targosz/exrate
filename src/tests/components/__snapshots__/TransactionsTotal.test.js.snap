// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - TransactionsTotal Should render TransactionsTotal correctly 1`] = `
<section
  className="total"
>
  <InfoBox
    data="583,07 â‚¬ / 1 826,36 PLN"
    description="Total (7 transactions)"
  />
</section>
`;
